### ISSUE TITLE
Linux Installation

### ISSUE THREAD

**Issue by iDJHost** [GwynethLlewelyn/restbot#11]
Instructions are a bit vague, I've downloaded the zip, extracted it to /home/restbot-master.
Have all the necessary dependencies as far I'm aware.

Mono -V
Mono JIT compiler version 6.12.0.107 (tarball Wed Dec  9 21:42:51 UTC 2020)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
mono-devel-6.12.0.107-0.xamarin.9.epel7.x86_64 already installed and latest version

msbuild -version
Microsoft (R) Build Engine version 16.6.0 for Mono
Copyright (C) Microsoft Corporation. All rights reserved.
msbuild-16.6+xamarinxplat.2020.10.22.18.31-0.xamarin.1.epel7.noarch already installed and latest version

dotnet -v
Welcome to .NET 6.0!
SDK Version: 6.0.400

nuget installed

Running nuget restore in primary directory:
The folder '/home/restbot-master' does not contain an msbuild solution or packages.config file to restore.
Running nuget restore in restbot-src directory:
The folder '/home/restbot-master/restbot-src' does not contain an msbuild solution or packages.config file to restore.

Any additional instructions or suggestions to get the application/program running?

**Comment by iDJHost** [comment]
Linux OS: CentOS 7 Minimal.
No GUI - CLI - Terminal only

**Comment by iDJHost** [comment]
Minimal Progression:

**dotnet msbuild -m**

MSBuild version 17.3.0+92e077650 for .NET
/usr/share/dotnet/sdk/6.0.400/Sdks/Microsoft.NET.Sdk/targets/Microsoft.PackageDependencyResolution.targets(267,5): error NETSDK1004: Assets file '/home/restbot-master/obj/project.assets.json' not found. Run a NuGet package restore to generate this file. [/home/restbot-master/RESTbot.csproj]

**nuget restore**

The folder '/home/restbot-master' does not contain an msbuild solution or packages.config file to restore.

**Comment by iDJHost** [comment]
Update:
I needed to run:

**dotnet restore**

Determining projects to restore...
/home/restbot-master/RESTbot.csproj : warning NU1603: RESTbot depends on LibreMetaverse (>= 1.9.20.453-gwyn) but LibreMetaverse 1.9.20.453-gwyn was not found. An approximate best match of LibreMetaverse 2.0.0.278 was resolved.
/home/restbot-master/RESTbot.csproj : warning NU1603: RESTbot depends on LibreMetaverse.StructuredData (>= 1.9.20.453-gwyn) but LibreMetaverse.StructuredData 1.9.20.453-gwyn was not found. An approximate best match of LibreMetaverse.StructuredData 2.0.0.278 was resolved.
/home/restbot-master/RESTbot.csproj : warning NU1603: RESTbot depends on LibreMetaverse.Types (>= 1.9.20.453-gwyn) but LibreMetaverse.Types 1.9.20.453-gwyn was not found. An approximate best match of LibreMetaverse.Types 2.0.0.278 was resolved.
/home/restbot-master/RESTbot.csproj : warning NU1603: RESTbot depends on LibreMetaverse.Utilities (>= 1.9.20.453-gwyn) but LibreMetaverse.Utilities 1.9.20.453-gwyn was not found. An approximate best match of LibreMetaverse.Utilities 2.0.0.278 was resolved.
  All projects are up-to-date for restore.

and then:

**dotnet msbuild -m**

MSBuild version 17.3.0+92e077650 for .NET
/home/restbot-master/RESTbot.csproj : warning NU1603: RESTbot depends on LibreMetaverse (>= 1.9.20.453-gwyn) but LibreMetaverse 1.9.20.453-gwyn was not found. An approximate best match of LibreMetaverse 2.0.0.278 was resolved.
/home/restbot-master/RESTbot.csproj : warning NU1603: RESTbot depends on LibreMetaverse.StructuredData (>= 1.9.20.453-gwyn) but LibreMetaverse.StructuredData 1.9.20.453-gwyn was not found. An approximate best match of LibreMetaverse.StructuredData 2.0.0.278 was resolved.
/home/restbot-master/RESTbot.csproj : warning NU1603: RESTbot depends on LibreMetaverse.Types (>= 1.9.20.453-gwyn) but LibreMetaverse.Types 1.9.20.453-gwyn was not found. An approximate best match of LibreMetaverse.Types 2.0.0.278 was resolved.
/home/restbot-master/RESTbot.csproj : warning NU1603: RESTbot depends on LibreMetaverse.Utilities (>= 1.9.20.453-gwyn) but LibreMetaverse.Utilities 1.9.20.453-gwyn was not found. An approximate best match of LibreMetaverse.Utilities 2.0.0.278 was resolved.
RESTbot -> /home/restbot-master/restbot-bin/net6.0/RESTbot.dll
Asset files copied.

**Success?

systemctl start aaa_restbot.service
systemctl status -l aaa_restbot.service**

● restbot.service - RESTbot
   Loaded: loaded (/etc/systemd/system/aaa_restbot.service; enabled; vendor preset: disabled)
   Active: active (running) since Tue 2022-09-06 20:07:33 EDT; 8s ago
  Process: 14372 ExecStart=/usr/bin/screen -dmS RESTbot /home/restbot-master/restbot-bin/net6.0/RESTbot (code=exited, status=0/SUCCESS)
 Main PID: 14373 (screen)
   CGroup: /system.slice/aaa_restbot.service
           ├─14373 /usr/bin/SCREEN -dmS RESTbot /home/restbot-master/restbot-bin/net6.0/RESTbot
           └─14374 /home/restbot-master/restbot-bin/net6.0/RESTbot

**Sep 06 20:07:33 systemd[1]: Starting RESTbot...
Sep 06 20:07:33 systemd[1]: Started RESTbot.**

Appears to be running now

**Comment by iDJHost** [comment]
OK, I have successfully logged in an avatar.

Just wanted to let you know, I "had" to use my MD5 password.

You can probably update this in the wiki:
"Second Life, however, mumbles and grumbles and ultimately refuses to use your MD5'd password. Why that happens is completely beyond my knowledge and understanding."

To something like this:
To use your MD5'd password for Second Life, you will need to use the escape character **"\"** for example:
\$1\$12345678901234567890123456789012 (this is not a real password).

```
curl http://localhost:9080/establish_session/<RESTbot password> -d first=Avatar -d last=Resident -d pass=\$1\$12345678901234567890123456789012
```
Will successfully work, given all the information is correct.

**Comment by iDJHost** [comment]
You can close this after you've had a chance to read it and maybe update documentation.

**Comment by GwynethLlewelyn** [comment]
Thanks for the extensive testing! Aye, I have to take a serious look at all the documentation and fix what's missing...

**Comment by iDJHost** [comment]
I think for the most part you got the installation documentation right, I didn't think to use the Mac instructions prior to submitting the issue. (completely my fault for rushing) However, I did find that I was unable to resolve the issue using `nuget restore` I had to use `dotnet restore` instead and then run `dotnet msbuild -m` to finalize the "installation" and get it running.

**Comment by GwynethLlewelyn** [comment]
Exactly, you need to use `dotnet` for everything — this _is_ a .NET application, after all (no need of the Mono runtime!).

**Comment by GwynethLlewelyn** [comment]
@all-contributors please add @iDJHost for tests
